Community/network_manager/network_manager_page.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 BlueCat Networks. All rights reserved.
+
+# Various Flask framework items.
+import os
+import sys
+from flask import url_for, redirect, render_template, flash, g
+from bluecat import route, util
+import config.default_config as config
+from main_app import app
+from .network_manager_form import GenericFormTemplate
+
+
+def module_path():
+    encoding = sys.getfilesystemencoding()
+    return os.path.dirname(os.path.abspath(str(__file__, encoding)))
+
+
+@route(app, '/network_manager/network_manager_endpoint')
+@util.workflow_permission_required('network_manager_page')
+@util.exception_catcher
+def network_manager_network_manager_page():
+    form = GenericFormTemplate()
+    u = g.user.get_api().get_user(g.user.get_username())
+    groups = u.get_user_groups()
+    has_block = False
+    for group in groups:
+        bv = g.user.get_api().get_by_object_types(group.name, ['IP4Block'])
+        has_block=False
+        for b in bv:
+            has_block = True
+            break
+    if has_block == False:
+        flash("You don't have any network space allocated to your group")
+    return render_template(
+        'network_manager_page.html',
+        form=form,
+        text=util.get_text(module_path(), config.language),
+        options=g.user.get_options(),
+    )
+
+@route(app, '/network_manager/form', methods=['POST'])
+@util.workflow_permission_required('network_manager_page')
+@util.exception_catcher
+def network_manager_network_manager_page_form():
+    form = GenericFormTemplate()
+    if form.validate_on_submit():
+        block = None
+        u = g.user.get_api().get_user(g.user.get_username())
+        groups = u.get_user_groups()
+        for group in groups:
+            bv = g.user.get_api().get_by_object_types(group.name, ['IP4Block'])
```

Comment:
get_by_object_types uses searchByObjectTypes underneath it and doesn't do any filtering by configuration. If you only have one configuration or only one block in the BAM with the group name then it is fine. Otherwise you run the risk of finding more than one block with the same name and there is no guarantee that the first block returned will be the one you actually want.

Reply:
If you think customers won't actually run into this issue then it is fine. Maybe just a note to not have more than one block with the same group name in multiple configurations.